### PR TITLE
TECH-981 Switch to async local storage

### PR DESCRIPTION
**Pull request by bevy-tech**

### CODE REVIEW

File: src/tracing/index.ts [hunk 1]
```diff
@@ -2,21 +2,28 @@ import axios from "axios";
 import {randomUUID} from "crypto";
 import { Request, Response, NextFunction } from 'express';
 
-const createNamespace = require('continuation-local-storage').createNamespace;
-const ns = createNamespace('logger-context');
+const { AsyncLocalStorage } = require('node:async_hooks');
```
Comment (dwielocha) [inline src/tracing/index.ts]: Why not import { AsyncLocalStorage } from 'node:async_hooks'; ?

File: src/tracing/index.ts [hunk 2]
```diff
@@ -2,21 +2,28 @@ import axios from "axios";
...
-export const getTraceHeader = () => ns.get(traceHeaderKey);
+export const getTraceId = () => `projects/${project}/traces/${als.getStore()}`;
+export const getTraceHeader = () => als.getStore();
 export const expressTracingMiddleware = (req: Request, res: Response, next: NextFunction) => {
+
+    const header: string | undefined = req.header(headerName);
+    const chunks: string[] | undefined = header?.split(';') ?? [];
+
+    als.run(chunks?.find(() => true), () => {
+        next();
```
Comment (dwielocha) [inline src/tracing/index.ts:22]: Soo.. the middleware execution flow is stopped until the callback is resolved? 🤔

**Review by dwielocha — APPROVED** [review]

File: src/tracing/index.ts (hunk 2 above)
Comment (bevy-tech) [reply src/tracing/index.ts:22]: I'm not following... here we are just calling next immediately to preserve the current async context

File: src/tracing/index.ts (hunk 1 above)
Comment (bevy-tech) [reply src/tracing/index.ts]: fair

**Pull request merged**